src/components/CohortBuilder/Results.js

```diff
@@ -166,92 +167,110 @@ const Results = ({
       const hasNoFile = filesCardinality === 0;
       const hasNoCohortQuery = isEmpty(sqon.content);
 
-      const showDetailsHeader = () => {
-        if (hasNoCohortQuery) {
-          return <H2>All Data</H2>;
-        }
-        return (
-          <Fragment>
-            <H2>Cohort Results</H2>
-            <h3 className="cb-sub-heading" style={{ fontWeight: 'normal', marginRight: '10px' }}>
-              for Query {activeSqonIndex + 1}
-            </h3>
-          </Fragment>
-        );
-      };
+      const showDetailsHeader = () => (
+        <div className="cb-extra-actions-header">
+          {hasNoCohortQuery ? (
+            <H2>All Data</H2>
+          ) : (
+            <Fragment>
+              <H2>Cohort Results</H2>
+              <h3 className="cb-sub-heading" style={{ fontWeight: 'normal', marginRight: '10px' }}>
+                for Query {activeSqonIndex + 1}
+              </h3>
+            </Fragment>
+          )}
+        </div>
+      );
+
+      const extraActions = (
+        <div className="cb-extra-actions">
+          {showDetailsHeader()}
+          {isLoading ? (
+            <div className={'cb-summary-is-loading'}>
+              <Spin size={'small'} />
+            </div>
+          ) : (
+            <div className="cb-summary">
+              <div className="cb-summary-entity">
+                <h3 className="cb-sub-heading">
+                  <DemographicIcon width="14px" height="17px" />
```

Comment:
do we have antd equivalent for icons and images?

Reply:
shuffled code